src/main/java/com/example/everkan/registration/EmailValidator.java

```diff
@@ -0,0 +1,15 @@
+package com.example.everkan.registration;
+
+import org.springframework.stereotype.Service;
+
+import java.util.function.Predicate;
+
+@Service
+public class EmailValidator implements Predicate<String> {
+
+    @Override
+    public boolean test(String s) {
+        // TODO: Reqex to validate email
+        return true;
+    }
+}
```

Comment:
You could also use the validation framework to check mail addresses.

They work via annotations and are e.g. supported in request objects.

This dependency...

    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-validation</artifactId>
    </dependency>

... in combination with the tutorial https://www.baeldung.com/spring-boot-bean-validation might be a good starting point.

For emails addresses there is the annotation `javax.validation.constraints.Email`.

Reply:
Thanks for the tip. I will adopt it.